nextcord/polls.py

```diff
@@ -0,0 +1,441 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, AsyncIterator, List, Optional, Union, cast
+
+from .emoji import Emoji
+from .enums import PollLayoutType, try_enum
+from .errors import InvalidArgument
+from .iterators import answer_voters_iterator
+from .partial_emoji import PartialEmoji
+from .types.emoji import PartialEmoji as PartialEmojiPayload
+from .utils import MISSING, utcnow
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .abc import Snowflake
+    from .member import Member
+    from .message import Message
+    from .state import ConnectionState
+    from .types.polls import (
+        DAPIPollAnswer as DAPIPollAnswerPayload,
+        Poll as PollData,
+        PollAnswer as PollAnswerPayload,
+        PollAnswerCount as PollAnswerCountPayload,
+        PollCreateRequest as PollCreateRequestPayload,
+        PollMedia as PollMediaPayload,
+        PollResults as PollResultsPayload,
+    )
+    from .user import User
+
+    EmojiInputType = Union[Emoji, PartialEmoji, str]
+
+__all__ = (
+    "PollResults",
+    "PollMedia",
+    "Poll",
+    "PollAnswerCount",
+    "PollCreateRequest",
+    "PollAnswer",
+)
+
+
+def resolve_emoji(emoji: Union[Emoji, PartialEmoji]) -> PartialEmojiPayload:
+    if isinstance(emoji, Emoji):
+        return cast(PartialEmojiPayload, emoji._to_partial().to_dict())
+
+    if isinstance(emoji, PartialEmoji):
+        return cast(PartialEmojiPayload, emoji.to_dict())
+
+    raise InvalidArgument(f"Cannot parse `{emoji}` into an emoji to send.")
+
+
+class PollMedia:
+    """A common object that backs both the question and answer.
```

Comment:
I'm not too sure about this description, doesn't seem to describe the actual use very well.

Reply:
"Represents the data for a poll's question and answers." perhaps, I just think the wording of "backs" is that friendly to understand